src/publicapi.js

```diff
@@ -158,6 +158,10 @@ function getInterface(v) {
       if (this.__controller.blurred) this.__controller.cursor.hide().parent.blur();
       return this;
     };
+    _.empty = _.clear = function() {
+      this.select().__controller.backspace();
+      return this;
+    };
```

Comment:
Well if you do \MathQuillMathField[answer]{}, could you can do
outerMathFieldVariableNameWhatever.innerFields.answer.latex('')?

Han

On Fri, Mar 25, 2016 at 9:05 AM, Eoghan McIlwaine <notifications@github.com>
wrote:

> In src/publicapi.js
> <https://github.com/mathquill/mathquill/pull/553#discussion_r57458178>:
>
> > @@ -158,6 +158,10 @@ function getInterface(v) {
> >        if (this.__controller.blurred) this.__controller.cursor.hide().parent.blur();
> >        return this;
> >      };
> > +    _.empty = _.clear = function() {
> > +      this.select().__controller.backspace();
> > +      return this;
> > +    };
>
> Yeah, the .select() thing isn't ideal - we must come up with a better
> approach.
>
> @karoltarasiuk <https://github.com/karoltarasiuk> was there a reason for
> the double .empty() / .clear() naming? We only need one method name on
> our side.
>
> Just to add some context @laughinghan <https://github.com/laughinghan>:
> we added this method so we could clear just the currently focused
> \MathQuillMathField{} without affecting stuff around it. If there were no
> \MathQuillMathField{}s, we wouldn't need this method as we could just
> call .latex('').
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/mathquill/mathquill/pull/553/files/dcc29d74ecea4560ddee465c60b5c662b605c539#r57458178>
>